ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -159,3 +179,61 @@ impl IndexerStream for DatastreamServer {
         ))
     }
 }
+
+/// Builds the response for the raw datastream request. Partial batch is ok, i.e., a batch with transactions < 1000.
+fn raw_datastream_response_builder(
+    data: Vec<EncodedTransactionWithVersion>,
+    chain_id: u32,
+) -> RawDatastreamResponse {
+    RawDatastreamResponse {
+        response: Some(DatastreamProtoResponse::Data(TransactionsOutput {
+            transactions: data
+                .into_iter()
+                .map(|(encoded, version)| TransactionOutput {
+                    encoded_proto_data: encoded,
+                    version,
+                    ..TransactionOutput::default()
+                })
+                .collect(),
+        })),
+        chain_id,
+    }
+}
```

Comment:
make sure that the data is monotonically increasing. and return the latest version so that we don't have to do math.

Reply:
done, remove all the math from service.